### PR TITLE
refactor: installer

### PR DESCRIPTION
**Pull request by S0ly**
this branch aim to refactor the installer 

- [x] finish and merge #951 
- [x] refactor view
- [x] refactor functions
- [x] refactor forms

this refactor :

- rename install to installer
- split index into many many views and an index 
- split function into many functions
- split forms into many form
- remake the step system so that its a lot more easy to add or modify steps
- remake the error system so that error are more easy and better managed
- added a progress bar, and a footer
- added a return button and a revert system
- fix and remove some broken or useless code and clean most of the code
- updated the .env to be more logical and added documentation to it

this is a good start for a refactor but it may be even more upgraded

### CODE REVIEW

**Comment by S0ly** [comment]
this cannot be merged before #951 is merged

**Comment by 1day2die** [comment]
yooo wtf is this. interessting. though we are now in a lot of merge conflics due toi PR 951

**Comment by S0ly** [comment]
why the fuck there is still conflicts

**Comment by S0ly** [comment]
I cant fix them they these two files are deleted on my side they dont even exist

**Comment by S0ly** [comment]
tbh I dont know the what I have done

**Comment by S0ly** [comment]
but it seems to be resolved now, just maybe try to test it before

File: app/Settings/TicketSettings.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class TicketSettings extends Settings
 {
-    public bool $enabled;
+    public ?bool $enabled;
```
Comment (IceToast) [inline app/Settings/TicketSettings.php]: Why is this optional? A boolean should not be nullable but rather have a default value

**Review by IceToast — CHANGES_REQUESTED** [review]

File: app/Settings/TicketSettings.php (hunk 1 above)
Comment (S0ly) [reply app/Settings/TicketSettings.php]: I dont know this is not a change from me, its a change from jameskitt if I remember well on te original installer fix branch

Comment (S0ly) [reply app/Settings/TicketSettings.php]: yep commit [Fix wrong php typehint](https://github.com/Ctrlpanel-gg/panel/pull/972/commits/9772321a4d426f729cc90b2779ff8b783793339d)

@jameskitt616

Comment (IceToast) [reply app/Settings/TicketSettings.php]: Needs to be resolved, a boolean cannot be nullable.

Comment (S0ly) [reply app/Settings/TicketSettings.php]: fixed

**Review by IceToast — APPROVED** [review]

**Pull request merged**